src/integrationTest/java/uk/gov/hmcts/darts/testutils/PostgresIntegrationBase.java

```diff
@@ -30,13 +30,16 @@ public class PostgresIntegrationBase {
 
     @DynamicPropertySource
     static void configureProperties(DynamicPropertyRegistry registry) {
-
         registry.add("spring.datasource.url", POSTGRES::getJdbcUrl);
         registry.add("spring.datasource.username", POSTGRES::getUsername);
         registry.add("spring.datasource.password", POSTGRES::getPassword);
     }
 
     static {
+        // Keep an eye on this, if it needs to be increased as more tests utilise PostgresIntegrationBase then we may have a connection leak somewhere
+        final int serverMaxConnections = 50;
```

Comment:
Why do we need to do this when running a single threaded test? Just curious.

Reply:
There's a bit of a story to this! When I moved the `RetentionGetPolicyTypesServiceImplTest` over to use Postgres, the `./gradlew build` would consistently fail at the integration test stage with:

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Failed to initialize dependency 'flywayInitializer' of LoadTimeWeaverAware bean 'entityManagerFactory': Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Unable to obtain connection from database: FATAL: sorry, too many clients already
---------------------------------------------------------------------------------
SQL State  : 53300
Error Code : 0
Message    : FATAL: sorry, too many clients already
```

As I understand it, there are a two aspects to the connection config:
- `spring.datasource.hikari.maximum-pool-size` defines the maximum number of connections that our test client can impose against the DB server. This value is set in `application.yaml` to 50.
- `max-connections` can be configured on the DB side, if the number of client connections exceeds this value then we'll see the above error. The value defaults to 100 for the `postgres:alpine-15` image.

To avoid the "too many clients" issue, this PR:
- Sets `maximum-pool-size` to 10, which is a downgrade from the 50 configured for the main app. This seems reasonable (and maybe excessive), as you say the tests are single threaded so we shouldn't ever need 50 connections (or even 10?).
- Sets `max-connections` DB-side to 50, which is a downgrade from the 100 configured by default. Interestingly, this is about as low as I can go without seeing the exception above. I do know that Postgres expects some (non-client) connections to be reserved for internal use, so maybe that's why the extra overhead (of 40??) connections is needed, but not clear.

There remains the root cause of why I had to do this, and I'm not sure on that. The fact the test suite fails when I added a new test class using `PostgresIntegrationBase` leads me to believe the new test has increased the connection demand on the db server, which shouldn't happen. If we truely do have a connection leak somewhere then we should see this problem again when usage of `PostgresIntegrationBase` becomes more common - this is why I've lowered `max-connections` to the lowest tolerable value, so that we get that feedback relatively quickly.

Reply:
ah ok makes sense to me. Like you say ideally we would work out why we need to do this but realise thats outside the scope of this ticket